packages/blocks/src/api/factory.js

```diff
@@ -134,11 +135,17 @@ export function createBlocksFromInnerBlocksTemplate(
 export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
 	const clientId = uuid();
 
+	const blockType = getBlockType( block.name );
+	let attributes = block.attributes;
+	if ( blockType ) {
+		attributes = pick( attributes, Object.keys( blockType.attributes ) );
```

Comment:
Welp, all of the E2E tests pass which is encouraging! 😀

I think my main hesitation with this approach is that it might be unexpected to developers that `createBlock()` and `cloneBlock()` behave differently.

Mind you I noticed you're setting `__internalWidgetId` _after_ `createBlock()` is called, though, so maybe we could make this change to `createBlock()` too and then that way the two functions will be consistent?

Another consideration is that technically this is a breaking change to a public API. It might be okay though as it's a change to behaviour that was never properly defined.

What do you think of this @youknowriad?

Reply:
Good call! `createBlock` already has something similar: `sanitizedAttributes`, maybe I can try to extract it to a util and use it on both `createBlock` and `cloneBlock`.